docs/architecture/metadata/structuring-yaml.md

```diff
@@ -0,0 +1,171 @@
+
+Multi-line strings are often sources of confusion. YAML supports two primary styles for writing them:
+
+```yaml
+my_var_1:
+  description: |-
+    The first line
+    and the second
+
+    Third line after line break
+```
+
+```yaml
+my_var_1:
+  description: >-
+    Just a
+    single line
+
+    Second line
+
+
+    Third line after line break
+```
+
+It's up to you whether you want to fit the entire text on the single line and use `>-` or use `|-` and break the lines.
+
+Using `-` after `|` or `>` removes whitespaces at the beginning and end of the string. This is almost always what you want.
+
+
+## Anchors & aliases
+
+[Anchors and aliases](https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/) are a native YAML functionality and they can be used to reduce repetition.
+
+Typically we define a special section called `definitions:` at the very top of the file and then we use aliases to refer to these definitions.
+
+An example that reuses `attribution` and `description_key`
+
+```yaml
+definitions:
+  attribution: &attribution Fishcount (2018)
+  description_key_common: &description_key_common
+    - First line.
+    - Second line.
+  description_key_individual:
+    - &third_line Third line.
+
+tables:
+  my_table:
+    variables:
+      my_var_1:
+        description_key: *description_key_common
+        presentation:
+          attribution: *attribution
+      my_var_2:
+        description_key:
+          - *description_key_common
+          - *third_line
+```
+
+
+## Common fields for all indicators
+
+!!! warning "In progress."
+
+To avoid repetition for all indicators, you can use a special section called `common:` under `definitions:`. This section is then merged with sections of all indicators. Using this saves you from repeating the same aliases in indicators.
+
+```yaml
+definitions:
+  common:
+    description_key:
+      - First line.
+      - Second line.
+    presentation:
+      grapher_config:
+        selectedEntityNames:
+          - Germany
+          - Italy
+        topic_tags:
+          - Energy
+
+tables:
+  my_table:
+    variables:
+      my_var_1:
+        # Final description will be First, Second, Third line
+        description_key:
+          - Third line.
+        presentation:
+          # Final tags will be Internet, Energy
+          topic_tags:
+            - Internet
+```
+
+
+## Dynamic YAML
+
+Anchors and aliases have limitations. One of the main ones is that you cannot use it for in-line text. That's why we've added support for [dynamic-yaml](https://github.com/childsish/dynamic-yaml) which lets you write YAML files likes this:
+
+```yaml
+definitions:
+  additional_info: |-
+    You should also know this.
+
+tables:
+  my_table:
+    variables:
+      my_var_1:
+        description: |-
+          This is a description.
+          { additional_info }
+```
+
+There are also special variables like `{ TODAY }` that can be used for automatically updated datasets.
```

Comment:
I think we should never use `TODAY` (except for scripts that run literally every day), because every random day we run the step it will change the file and make the file and its dependencies dirty, right? If this is correct, and we should discourage this use, maybe we should remove this from the docs.

Reply:
> it will change the file and make the file and its dependencies dirty, right?

Not really. The file checksum is calculated using the string "TODAY", not the actual date, so TODAY will only change if dependencies get updated. We use this for covid automatic updates.